### ISSUE TITLE
Add support for assembly language for foreign_stubs

### ISSUE THREAD

**Issue by samoht** [ocaml/dune#3342]
The doc for `foreign_stubs` says: 

> language specifies the source language, where c means C and cxx means C++. In future, more languages may be supported.

It would be nice to add support for `.S` files too ; generally they can just be passed to the underlying C compiler just fine ; I'm not fully sure of what should be the flags. I think keeping the flags as C by default should generally work.

**Comment by diml** [comment]
Indeed. That requires #3346

**Comment by rgrinberg** [comment]
@gridbugs do you think this is still worth doing given that we have custom objects?

**Comment by gridbugs** [comment]
I don't think this is worth doing anymore as one can now compile the assembly files to object files with a custom rule and then include them with `(extra_objects ...)`. For posterity, the relevant part of the documentation is https://dune.readthedocs.io/en/stable/concepts.html?highlight=extra_objects#extra-objects-1
